new/docs/design/01-new-cloudware-plans.md

```diff
@@ -0,0 +1,358 @@
+  has deployed this `Node`). For instance, a `${node_name}__id` `Output` will
+  be required to perform future actions to interact with the `Node` on the
+  cloud platform (like the `power` commands, described below), while a
+  `${node_name}__ip` action may later be used to provide a `console` command to
+  easily SSH to a `Node`.
+
+  Note the double-underscore separating the two parts of the name of these
+  special `Node` `Output`s: this was chosen as (I believe) underscores should
+  always be valid in output names for all platforms we will initially/may later
+  want to support, and we may want to use single underscores to provide normal
+  separation within output names.  Also note that, since a single node can only
+  be deployed to a single location at a time, each `Node`'s `name` uniquely
+  identifies this `Node` within the database.
+
+- This database will represent the state of everything (that we want to track)
+  deployed by the current Cloudware installation at the current time. If
+  needed, we can later switch to using soft deletes if we want to retain a
+  history of all past things deployed as well.
+
+## Simple templating system
+
+Cloudware will use templates to deploy to each of the cloud platforms that it
+supports. Cloudware will not be prescriptive about how these templates should
+be created; ultimately in practise however it is likely that
+[Underware](https://github.com/alces-software/underware) (following various
+incoming changes to this) will often be used to create these templates.
+
+As such, most complex aspects of creating Cloudware templates will be beyond
+the scope of Cloudware itself, however Cloudware will need some simple
+templating functionality in order to support using the `value`s of `Output`s
+from previous `Deployment`s in later `Deployment`s. Two initial considerations
+for choosing this templating system are as follows:
+
+1. The Cloudware templating system should be able to unobtrusively live
+   alongside the Underware templating system. For example, it should be
+   possible for a template to be rendered by Underware to automate complex
+   aspects of constructing the template like looping through various values,
+   having various conditional sections depending on the current Underware
+   configuration, rendering partial templates within the main template etc.,
+   and then for this Underware-rendered template to also be rendered by
+   Cloudware, without special consideration needing to be made in either of
+   these stages to support the other stage (like escaping ERB tags etc.).
+
+2. The Cloudware templating system should be intentionally simplistic and just
+   needs to support the insertion of `Output` values in otherwise static
+   templates  - anything more complex than this is beyond the scope of
+   Cloudware (and is the responsibility of Underware or however else templates
+   are being created).
+
+Given these requirements, [Mustache](https://github.com/mustache/mustache) may
```

Comment:
Mustache may not be a good idea in this case as `{{name}}` is to similar to `json` and will make the templates hard to write